spec/system/scroll_manager_service_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+describe "Ember route-scroll-manager service", type: :system do
+  before do
+    Fabricate(:admin)
+    50.times { Fabricate(:post) }
+  end
+
+  let(:discovery) { PageObjects::Pages::Discovery.new }
+  let(:topic) { PageObjects::Pages::Topic.new }
+
+  def current_scroll_y
+    page.execute_script <<~JS
+      return window.scrollY
+    JS
+  end
+
+  it "scrolls to top when navigating to new routes, and remembers scroll position when going back" do
+    visit("/")
+    expect(page).to have_css("body.navigation-topics")
+    expect(discovery.topic_list).to have_topics
+
+    page.execute_script <<~JS
+      document.querySelectorAll('.topic-list-item')[10].scrollIntoView(true);
+    JS
+
+    topic_list_scroll_y = current_scroll_y
+    try_until_success { expect(topic_list_scroll_y).to be > 0 }
+
+    find(".sidebar-section-link[data-link-name='all-categories']").click
+
+    expect(page).to have_css("body.navigation-categories")
+
+    try_until_success { expect(current_scroll_y).to eq(0) }
+
+    page.driver.browser.navigate.back
```

Comment:
I think you can just do `page.go_back` now https://www.rubydoc.info/gems/capybara/Capybara%2FSession:go_back